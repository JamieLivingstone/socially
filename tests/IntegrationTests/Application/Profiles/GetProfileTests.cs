using System.Linq;
using System.Threading.Tasks;
using Application.Common.Exceptions;
using Application.Profiles.Queries.GetProfile;
using IntegrationTests.Application.TestUtils;
using NUnit.Framework;
using Snapshooter.NUnit;

namespace IntegrationTests.Application.Profiles;

public class GetProfileTests : TestBase
{
  [Test]
  public void GivenProfileDoesNotExist_ThrowsNotFoundException()
  {
    var query = new GetProfileQuery
    {
      Username = "does-not-exist",
    };

    async Task Handler() => await SendAsync(query);

    Assert.ThrowsAsync(typeof(NotFoundException), Handler);
  }

  [Test]
  public async Task GivenProfileExists_ReturnsProfileDto()
  {
    var query = new GetProfileQuery
    {
      Username = Seed.Persons().First().Username,
    };

    var profile = await SendAsync(query);

    Snapshot.Match(profile);
  }
}
