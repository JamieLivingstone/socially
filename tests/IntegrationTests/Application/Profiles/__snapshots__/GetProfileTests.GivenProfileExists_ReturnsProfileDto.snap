{
  "Name": "Liam Hale",
  "Username": "liam",
  "CommentsCount": 2,
  "FollowersCount": 0,
  "PostsCount": 1,
  "Following": false
}
